plugins/tableselection/plugin.js

```diff
@@ -1008,12 +1009,35 @@
 							ranges[ 0 ].moveToElementEditablePosition( firstCell );
 						}
 
+						if ( keystroke === 13 ) {
+							insertNewLineInsideFirstRange( editor, evt.data, ranges );
```

Comment:
It feels dirty to reimplement this logic… Isn't there any sensible way to not destroy table, but at the same time – delegate <kbd>Enter</kbd> handling to our existing mechanics?